src/clients/c/tb_client/packet.zig

```diff
@@ -31,31 +33,31 @@ pub const Packet = extern struct {
         invalid_data_size,
     };
 
-    /// Thread-safe stack optimized for 1 consumer (io thread) and N producers (client threads),
-    /// `push` uses a spin lock, and `pop` is not thread-safe.
-    pub const SubmissionStack = struct {
-        pushed: Atomic(?*Packet) = Atomic(?*Packet).init(null),
-        popped: ?*Packet = null,
-
-        pub fn push(self: *SubmissionStack, packet: *Packet) void {
-            var pushed = self.pushed.load(.monotonic);
-            while (true) {
-                packet.next = pushed;
-                pushed = self.pushed.cmpxchgWeak(
-                    pushed,
-                    packet,
-                    .release,
-                    .monotonic,
-                ) orelse break;
-            }
+    /// Thread-safe FIFO.
+    pub const SubmissionQueue = struct {
+        fifo: FIFOType(Packet) = .{
+            .name = null,
+            .verify_push = builtin.is_test,
+        },
+        mutex: std.Thread.Mutex = .{},
```

Comment:
Good call to slap a mutex onto it! This is control plane, so we don't need lock-free here!